src/psyclone/tests/psyir/nodes/call_test.py

```diff
@@ -628,6 +629,539 @@ def test_call_get_callees_local(fortran_reader):
+
+            print(f" - Found matching argument list: {arg_idx_list}")
+
+            assert len(arg_idx_list) == 3
+            assert arg_idx_list[0] == 0
+            assert arg_idx_list[1] == 1
+            assert arg_idx_list[2] == 2
+
+            assert result is routine_foo_c
+            print(" - Passed subtest foo_c[0]")
+
+        if 1:
+            print("Subtest foo_c[1]:")
+
+            call_foo_c: Call = routine_main.walk(Call)[6]
+            assert call_foo_c.routine.name == "foo"
+
+            arg_idx_list = []
+            result: Routine = call_foo_c.get_callee(
+                                ret_arg_match_list=arg_idx_list)
+
+            print(f" - Found matching argument list: {arg_idx_list}")
+
+            assert len(arg_idx_list) == 2
+            assert arg_idx_list[0] == 1
+            assert arg_idx_list[1] == 0
+
+            assert result is routine_foo_c
+            print(" - Passed subtest foo_c[1]")
+
+        if 1:
+            print("Subtest foo_c[2]:")
+
+            call_foo_c: Call = routine_main.walk(Call)[7]
+            assert call_foo_c.routine.name == "foo"
+
+            arg_idx_list = []
+            result: Routine = call_foo_c.get_callee(
+                                ret_arg_match_list=arg_idx_list)
+
+            print(f" - Found matching argument list: {arg_idx_list}")
+
+            assert len(arg_idx_list) == 3
+            assert arg_idx_list[0] == 1
+            assert arg_idx_list[1] == 0
+            assert arg_idx_list[2] == 2
+
+            assert result is routine_foo_c
+            print(" - Passed subtest foo_c[2]")
+
+
+def test_call_get_callee_7_matching_arguments_not_found(fortran_reader):
+    """
+    Trigger error that matching arguments were not found
+    """
+    code = """
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    ! Use name 'd' which doesn't exist
+    call foo(e, f, d=g)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b, c
+  end subroutine
+
+end module some_mod"""
+
+    psyir = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = psyir.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+
+    try:
+        call_foo.get_callee()
+
+    except CallMatchingArgumentsNotFound as err:
+        assert "No matching routine found for 'call foo(e, f, d=g)" in str(err)
+        print("Success! Exception triggered (as expected)")
+        return
+
+    assert False, (
+        "This should have triggered an error since there"
+        "are more arguments in the call than in the routine"
+    )
+
+
+def test_call_get_callee_8_arguments_not_handled(fortran_reader):
+    """
+    Trigger error that matching arguments were not found
+    """
+    code = """
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f
+    ! Use name 'd' which doesn't exist
```

Comment:
I don't understand this comment. Perhaps 'omit the required third argument'?

Reply:
I extended the comment:

    ! Use named argument 'd', which doesn't exist
    ! to trigger an error when searching for the matching routine.

Reply:
OK. Seems that this got lost.

Reply:
Sorry Martin, I might be being dense here but L1250 still talks about 'd' but neither the call nor the routine definition have a 'd'?

Reply:
Indeed, 'd' didn't make sense here.
I hope I treated this now correctly.